src/platform/silabs/wifi/DnssdImpl.cpp

```diff
@@ -0,0 +1,172 @@
+#include "lib/dnssd/platform/Dnssd.h"
+#include "platform/CHIPDeviceLayer.h"
+extern "C" {
+#include "sl_mdns.h"
+}
+#include <lib/support/CHIPMem.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+using namespace ::chip::DeviceLayer;
+
+namespace chip {
+namespace Dnssd {
+
+// Global mDNS instance
+static sl_mdns_t gMdnsInstance;
+
+CHIP_ERROR ChipDnssdInit(DnssdAsyncReturnCallback initCallback, DnssdAsyncReturnCallback errorCallback, void * context)
+{
+    sl_mdns_configuration_t config = { .protocol = SL_MDNS_PROTO_UDP, .type = SL_IPV6_VERSION, .host_name = "chip-device.local" };
+
+    sl_status_t status = sl_mdns_init(&gMdnsInstance, &config, nullptr);
+    if (status != SL_STATUS_OK)
+    {
+        ChipLogError(DeviceLayer, "Failed to initialize mDNS: 0x%lx", status);
+        if (errorCallback)
+        {
+            errorCallback(context, CHIP_ERROR_INTERNAL);
+        }
+        return CHIP_ERROR_INTERNAL;
+    }
+    status = sl_mdns_add_interface(&gMdnsInstance, SL_NET_WIFI_CLIENT_INTERFACE);
+    if (status != SL_STATUS_OK)
+    {
+        return CHIP_ERROR_INTERNAL;
+    }
+
+    if (initCallback)
+    {
+        initCallback(context, CHIP_NO_ERROR);
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+void ChipDnssdShutdown()
+{
+    sl_status_t status = sl_mdns_deinit(&gMdnsInstance);
+    if (status != SL_STATUS_OK)
+    {
+        ChipLogError(DeviceLayer, "Failed to deinitialize mDNS: %ld", status);
+    }
+}
+
+const char * GetProtocolString(DnssdServiceProtocol protocol)
+{
+    return protocol == DnssdServiceProtocol::kDnssdProtocolTcp ? "_tcp" : "_udp";
+}
+
+sl_mdns_service_t mdnsService = {};
+std::string serviceMessage;
+
+CHIP_ERROR ChipDnssdPublishService(const DnssdService * service, DnssdPublishCallback callback, void * context)
```

Comment:
Function is missing several input validation which should be present

- pointer validation
- callback validation and usage
- protocol support

Reply:
Done